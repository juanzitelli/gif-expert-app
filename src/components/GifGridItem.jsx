import React from 'react'
import PropTypes from 'prop-types'


const GifGridItem = ({title,url}) =>

	<div className="card animate__animated animate__fadeIn">
		<img src={url} alt={title}/>
		<p>{title}</p>
	</div>

	

GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
}

GifGridItem.defaultProps = {
	title: "Default GIF title"
}

export default GifGridItem
