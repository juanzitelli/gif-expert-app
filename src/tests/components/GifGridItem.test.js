import GifGridItem from "../../components/GifGridItem";
import React from "react";
const { shallow } = require("enzyme");

describe("Pruebas en GifGridItem", () => {
  const test_title = "Some title";
  const test_url = "https://localhost/algo.jpg"

  const wrapper = shallow(<GifGridItem url={test_url} title={test_title} />);
  
  

  test("Debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe requerir tÃ­tulo", () => {
	const title = wrapper.find("p").text().trim();
	expect(title).toBe(test_title);
  });

  test("Debe requerir url", () => {
	const url = wrapper.find("img").prop("src").trim();
	expect(url).toBe(test_url);
	
  });

  test("Debe tener la clase animate__fadeIn", () => {
	const hasClassFadeIn = wrapper.find("div").hasClass('animate__fadeIn');
    expect(hasClassFadeIn).toBe(true);
  });
});
