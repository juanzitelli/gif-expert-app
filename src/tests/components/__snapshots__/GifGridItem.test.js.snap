// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Some title"
    src="https://localhost/algo.jpg"
  />
  <p>
    Some title
  </p>
</div>
`;
